hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NSSummaryEndpoint.java

```diff
@@ -128,8 +139,110 @@ public Response getDiskUsage(@QueryParam("path") String path,
+   * @param creationDate Filter for keys created after creationDate in "MM-dd-yyyy HH:mm:ss" string format.
+   * @param keySize Filter for Keys greater than keySize in bytes.
+   * @param startPrefix Filter for startPrefix path.
+   * @param limit Filter for limited count of keys.
+   * @param recursive listing out keys recursively for FSO buckets.
+   * @return the list of keys in below structured format:
+   * Response For OBS Bucket keys:
+   * ********************************************************
+   * {
+   *     "status": "OK",
+   *     "path": "/volume1/obs-bucket/",
+   *     "size": 73400320,
+   *     "sizeWithReplica": 81788928,
+   *     "subPathCount": 1,
+   *     "totalKeyCount": 7,
+   *     "lastKey": "/volume1/obs-bucket/key7",
+   *     "subPaths": [
+   *         {
+   *             "key": true,
+   *             "path": "key1",
+   *             "size": 10485760,
+   *             "sizeWithReplica": 18874368,
+   *             "isKey": true,
+   *             "replicationType": "RATIS",
+   *             "creationTime": 1712321367060,
+   *             "modificationTime": 1712321368190
+   *         },
+   *         {
+   *             "key": true,
+   *             "path": "key7",
+   *             "size": 10485760,
+   *             "sizeWithReplica": 18874368,
+   *             "isKey": true,
+   *             "replicationType": "EC",
+   *             "creationTime": 1713261005555,
+   *             "modificationTime": 1713261006728
+   *         }
+   *     ],
+   *     "sizeDirectKey": 73400320
+   * }
+   * ********************************************************
+   * @throws IOException
+   */
+  @GET
+  @Path("/listKeys")
+  @SuppressWarnings("methodlength")
+  public Response listKeysWithDu(@QueryParam("replicationType") String replicationType,
+                                 @QueryParam("creationDate") String creationDate,
+                                 @DefaultValue(DEFAULT_KEY_SIZE) @QueryParam("keySize") long keySize,
+                                 @DefaultValue(OM_KEY_PREFIX) @QueryParam("startPrefix") String startPrefix,
+                                 @DefaultValue(DEFAULT_FETCH_COUNT) @QueryParam("count") long limit,
+                                 @DefaultValue("false") @QueryParam("recursive") boolean recursive)
+      throws IOException {
+
+    if (startPrefix == null || startPrefix.length() == 0) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    DUResponse duResponse = new DUResponse();
+    if (!isInitializationComplete()) {
+      duResponse.setStatus(ResponseStatus.INITIALIZING);
+      return Response.ok(duResponse).build();
+    }
+    EntityHandler handler = EntityHandler.getEntityHandler(
+        reconNamespaceSummaryManager,
+        omMetadataManager, reconSCM, startPrefix);
+
+    Stats stats = new Stats(limit);
+
+    duResponse = handler.getListKeysResponse(stats, recursive);
```

Comment:
Instead of overloading DuReponse class, we can define new response class for listFile purpose and return only required information